react/tic-tac-toe/src/app/screens/Game/index.js

```diff
@@ -2,19 +2,98 @@ import React from 'react';
 
 import styles from './styles.module.scss';
 import Board from './components/Board';
+import { createSecureContext } from 'tls';
 
-function Game () {
-  return (
-    <div className={styles.game}>
-      <div className={styles.gameBoard}>
-        <Board />
-      </div>
-      <div className={styles.gameInfo}>
-        <div>{/* status */}</div>
-        <ol>{/* TODO */}</ol>
+class Game extends React.Component {
+  state = { 
+    history: [{ 
+      squares: Array(9).fill(null) 
+    }],
+    stepNumber: 0,
+    xIsNext: true
+  };
+
+  handleClick(i) {
+    const history = this.state.history.slice(0, this.state.stepNumber + 1);
+    const current = history[history.length - 1];
+    const squares = current.squares.slice();
+    if (calculateWinner(squares) || squares[i]) {
+      return;
+    }
+    squares[i] = this.state.xIsNext ? 'X' : 'O';
+    this.setState({ 
+      history: history.concat([{
+        squares: squares
+      }]),
+      stepNumber: history.length,
+      xIsNext: !this.state.xIsNext
```

Comment:
If your next state depends on the previous one you should pass an `updater` function that receives the prev state and returns the new one to `setState` instead of an object:

Something like:
```js
this.setState(prevState => ({
 history: history.concat([{
        squares: squares
      }]),
  stepNumber: history.length,
  xIsNext: !prevState.xIsNext
}));
```

This is because setState is async and could be call in batches, you can read more about setState here: https://reactjs.org/docs/react-component.html#setstate.